### PR TITLE
Add a test for unexpected activation triggered during space keydown driven focus management

### PR DESCRIPTION
**Pull request by muan**
Relevant spec: https://w3c.github.io/uievents/#event-flow-activation

This is targeting [a Firefox bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1220143) where when forwarding focus on space and enter keydown events, a unexpected click event is triggered on the forward target.

Here's a demo page for the buggy behavior: https://html-is.glitch.me/firefox-space-focus.html

Since this is my first WPT pull request, I want to keep it small– this test case only cover the Space key case where Firefox is the outlier. 

There might be a better place for this test file. I'd be happy to take any suggestions and make changes accordingly.

Many thanks to @hexcles & @jgraham for helping me in IRC.

---

Reference re `\ue00d`: https://bugzilla.mozilla.org/show_bug.cgi?id=1417955

### CODE REVIEW

File: uievents/order-of-events/focus-events/focus-management-expectations.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  async_test(function (t) {
+    from.addEventListener("keydown", function (event) {
+      if (event.key === " ") to.focus()
+    })
+
+    to.addEventListener("click", t.unreached_func("Button should not be clicked"))
+
+    from.focus()
+    new test_driver.Actions().keyDown("\ue00d").keyUp("\ue00d").send().then(() =>
+      t.done()
```
[the remark this answers is not in the document]
Comment (muan) [reply uievents/order-of-events/focus-events/focus-management-expectations.html]: Thanks @NavidZ. I've applied your suggestions by setting up a click sequence on a new button the end the test. Testing locally I'm still getting the same set of results 👍🏻 .

**Review by NavidZ — APPROVED** [review]

**Comment by Hexcles** [comment]
Is this safe to land now?

**Comment by muan** [comment]
@Hexcles Not sure who this question is for, but for what it's worth I've done all I can here. Is there something this needs to wait for?

There is this new comment on https://bugzilla.mozilla.org/show_bug.cgi?id=1220143#c12, which I've replied to.

**Comment by Hexcles** [comment]
Thanks, @muan. I'm merging this now. If anyone (including folks on the Bugzilla thread) has further comments, we can always follow up.

**Pull request merged**